speeduino/sensors.ino

```diff
@@ -625,11 +640,14 @@ byte getOilPressure()
 
     tempOilPressure = fastMap10Bit(tempReading, configPage10.oilPressureMin, configPage10.oilPressureMax);
     tempOilPressure = ADC_FILTER(tempOilPressure, 150, currentStatus.oilPressure); //Apply speed smoothing factor
-    //Sanity check
-    if(tempOilPressure > configPage10.oilPressureMax) { tempOilPressure = configPage10.oilPressureMax; }
-    if(tempOilPressure < 0 ) { tempOilPressure = 0; } //prevent negative values, which will cause problems later when the values aren't signed.
+  } 
+  else if (configPage10.oilPressureEnable == 2)
+  {
+    tempOilPressure = oilSensorOPStData.pressure*14.5037;
```

Comment:
If you need values in PSI you must do it from TS, not here.

Reply:
I've modified the calculations into the sensor ISR to produce the initial value in PSI scale as it will provide better resolution using ints and as it matches with the existing oilpressure scale.